src/main/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitChecker.java

```diff
@@ -152,33 +153,41 @@ public String getDisplayName() {
         return displayName;
     }
 
-    public static void configureChecker(TaskListener listener, GitHub gitHub) {
-        configureChecker(GitHubConfiguration.get().getApiRateLimitChecker(), listener, gitHub);
+    public static void configureChecker(@NonNull TaskListener listener, @NonNull GitHub gitHub) {
+        configureChecker(listener, gitHub.getApiUrl());
     }
 
-    public static LocalChecker configureChecker(@NonNull ApiRateLimitChecker apiChecker, @NonNull TaskListener listener, @NonNull GitHub gitHub) {
-        LocalChecker checker = apiChecker.getChecker(listener, gitHub.getApiUrl());
+    private static void configureChecker(TaskListener listener, String apiUrl) {
+        LocalChecker checker = GitHubConfiguration.get().getApiRateLimitChecker().getChecker(listener, apiUrl);
         localRateLimitChecker.set(checker);
-        return checker;
     }
 
-    static void resetChecker() {
-        localRateLimitChecker.set(null);
+    /**
+     * For test purposes only.
+     */
+    static LocalChecker getLocalChecker() {
+        return localRateLimitChecker.get();
     }
 
     /**
-     * This method is the old code path for rate limit checks
-     *
-     * It has been slowly refactored until it almost matches the behavior of the GitHubRateLimitChecker.
+     * For test purposes only.
      */
-    public void checkApiRateLimit(TaskListener listener, GitHub gitHub) throws IOException, InterruptedException {
-        LocalChecker currentChecker = configureChecker(this, listener, gitHub);
+    static void resetLocalChecker() {
+        localRateLimitChecker.set(null);
+    }
 
-        long count = 0;
-        while (currentChecker.checkRateLimit(
-                count == 0 ? gitHub.rateLimit().getCore() : gitHub.getRateLimit().getCore(),
-                count++)) {
-            Thread.sleep(1000);
+    static final class RateLimitCheckerAdapter extends RateLimitChecker {
```

Comment:
I went with an adapter that chooses which strategy to use for each thread based on the system setting at time when thread is configured.  This means we don't need to change GitHub instances in order to pick up changes to the system rate limit checking strategy.